fix: add contentStyle prop to Menu

jaulz opened the pull request:
<!-- Please provide enough information so that others can review your pull request. -->
<!-- Keep pull requests small and focused on a single change. -->

### Motivation

This will allow us to apply inner styles which could be useful if you want to have a Menu for a TextInput and don't want to have rounded corners at the top for example.

### Test plan

n/a

satya164 commented:
`theme={{ roundness: 0 }}` ?

jaulz commented:
That would also clear the bottom corners.

wojteg1337 commented:
I'm not sure, but adding new prop would break current snapshots. If all tests pass it's fine for me :)

jaulz commented:
@wojteg1337 actually, adding this new prop did not break any snapshots as long as it was undefined and all test passed. However, I added a new test case and also reworked the existing tests because they were not really working as they did not cover the actual menu but only the anchor.
